contracts/resolvers/DelegatableResolver.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity >=0.8.4;
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import "./profiles/ABIResolver.sol";
+import "./profiles/AddrResolver.sol";
+import "./profiles/ContentHashResolver.sol";
+import "./profiles/DNSResolver.sol";
+import "./profiles/InterfaceResolver.sol";
+import "./profiles/NameResolver.sol";
+import "./profiles/PubkeyResolver.sol";
+import "./profiles/TextResolver.sol";
+import "./profiles/ExtendedResolver.sol";
+import "./Multicallable.sol";
+import "./IDelegatableResolver.sol";
+
+/**
+ * A delegated resolver that allows the resolver owner to add an operator to update records of a node on behalf of the owner.
+ * address.
+ */
+contract DelegatableResolver is
+    Ownable,
+    Multicallable,
+    ABIResolver,
+    AddrResolver,
+    ContentHashResolver,
+    DNSResolver,
+    InterfaceResolver,
+    NameResolver,
+    PubkeyResolver,
+    TextResolver,
+    ExtendedResolver
+{
+    using BytesUtils for bytes;
+
+    // Logged when an operator is added or removed.
+    event Approval(
+        bytes32 indexed node,
+        address indexed operator,
+        bytes name,
+        bool approved
+    );
+
+    error NotAuthorized(bytes32 node);
+
+    constructor(address owner) {
+        operators[bytes32(0)][owner] = true;
+        transferOwnership(owner);
+    }
+
+    //node => (delegate => isAuthorised)
+    mapping(bytes32 => mapping(address => bool)) operators;
+
+    /*
+     * Check to see if the operator has been approved by the owner for the node.
+     * @param name The ENS node to query
+     * @param offset The offset of the label to query recursively. Start from the 0 position and kepp adding the length of each label as it traverse. The function exits when len is 0.
+     * @param operator The address of the operator to query
+     * @return node The node of the name passed as an argument
+     * @return authorized The boolean state of whether the operator is approved to update record of the name
+     */
+    function getAuthorizedNode(
```

Comment:
nit, `isAuthorised()` is british spelling, but `getAuthorizedNode()` is american. We should pick one for this contract, since most of ENS already has british spelling, I'd stick with that!